PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -590,9 +1054,29 @@ void ProjectPanel::notified(LPNMHDR notification)
 					}
 					TreeView_SetItem(_treeView.getHSelf(), &tvItem);
 				}
+				else if (nt == nodeType_baseDir)
+				{
+					ProjectPanelData& data = *reinterpret_cast<ProjectPanelData*>(tvnotif->item.lParam);
+
+					// a text was added for base directory - give it its own name
+					if (tvnotif->item.pszText && *tvnotif->item.pszText)
+					{
+						data._userLabel = tvnotif->item.pszText;
+					}
+					else
+					{
+						data._userLabel.clear();
+						generic_string newFolderLabel = buildDirectoryName(data._filePath, data._filters);
+						wcsncpy(tvItem.pszText,newFolderLabel.c_str(), MAX_PATH-1);

```

Comment:
- Why use `wcsncpy` when passing a `TCHAR[]`? (instead of `_tcsncpy`)
- You should use `_tcsncpy_s`.